### PR TITLE
fix `QgsNetworkAccessManager::blockingGet( req )`

### PR DESCRIPTION
**Pull request by MorriganR**
If `QgsNetworkAccessManager::blockingGet( req )` is executed on the main thread of the application, then a very tricky scheme for waking up the main thread is used.

1. Generate the signal `authRequestOccurred( reply, auth )`
https://github.com/qgis/QGIS/blob/f5bd9ccaee537e39648dfdabc977a3aefb4af4d1/src/core/network/qgsnetworkaccessmanager.cpp#L539

2. We put the request `handleAuthRequest` in the queue
https://github.com/qgis/QGIS/blob/f5bd9ccaee537e39648dfdabc977a3aefb4af4d1/src/core/network/qgsnetworkaccessmanager.cpp#L679

3. Wake up the main thread
https://github.com/qgis/QGIS/blob/f5bd9ccaee537e39648dfdabc977a3aefb4af4d1/src/core/network/qgsblockingnetworkrequest.cpp#L212

The main problem is that when the main thread wakes up (point 3) there is no guarantee that the authorization request (point 2) has already been queued. In this case, the main thread of the application goes back to `authRequestBufferNotEmpty.wait( &waitConditionMutex )` and stays there forever.
https://github.com/qgis/QGIS/blob/f5bd9ccaee537e39648dfdabc977a3aefb4af4d1/src/core/network/qgsblockingnetworkrequest.cpp#L250
This behavior is successfully detected by CI tests.
https://github.com/qgis/QGIS/runs/3491730890?check_suite_focus=true

I'm not sure that this fix won't break anything, since I only used CI tests.

### CODE REVIEW

File: src/core/network/qgsblockingnetworkrequest.cpp [hunk 1]
```diff
@@ -247,7 +247,7 @@ QgsBlockingNetworkRequest::ErrorCode QgsBlockingNetworkRequest::doRequest( QgsBl
         waitConditionMutex.unlock();
         break;
       }
-      authRequestBufferNotEmpty.wait( &waitConditionMutex );
+      //authRequestBufferNotEmpty.wait( &waitConditionMutex );
```
Comment (rouault) [inline src/core/network/qgsblockingnetworkrequest.cpp:250]: maybe you could try the 2 arguments wait() with the mutex and a timeout (I believe in millisecond) : https://doc.qt.io/qt-5/qwaitcondition.html#wait-1
And add a comment in the code why that timeout is needed

Comment (MorriganR) [reply src/core/network/qgsblockingnetworkrequest.cpp:250]: I don't understand the real reasons, but it seems like the idea with `sleep` or modified `wait` is failing #7055 :(
maybe like this:
`
QgsApplication::instance()->processEvents( QEventLoop::WaitForMoreEvents );

**Comment by nyalldawson** [comment]
@MorriganR processEvents is a really dangerous call to make -- it has lots of unintended side effects

File: src/core/network/qgsblockingnetworkrequest.cpp [hunk 2]
```diff
@@ -256,7 +256,7 @@ QgsBlockingNetworkRequest::ErrorCode QgsBlockingNetworkRequest::doRequest( QgsBl
       if ( !threadFinished )
       {
         waitConditionMutex.unlock();
-
+        QThread::currentThread()->usleep( 100 );
         QgsApplication::instance()->processEvents();
```
Comment (troopa81) [inline src/core/network/qgsblockingnetworkrequest.cpp:260]: Why not waiting for the handleRequest to be taken care of with an eventloop like this 
```
QEventLoop loop;
connect ( QgsNetworkAccessManager::instance(), &QgsNetworkAccessManager::requestAuthDetailsAdded, &loop, &QEventLoop::quit );
loop.exec();
```

I understand with the PR #7055 that there is issues with QEventLoop in main thread, do you know why ?

**Comment by MorriganR** [comment]
I give up, my brain is hacked... :)
Ready to close this pull request as unsuccessful.
It seems that I have not been able to determine the true reason for the failure of the test, the statement in the first post of this PR is false!

With a high degree of certainty, I can only say that a very fast response from the http-server is required to reproduce the problem.

**Pull request closed without merging**